src/modules/networks/Networks.ts

```diff
@@ -25,152 +25,75 @@ function toCollection<T, K>(
   return result;
 }
 
-function localeCompareWithPriority(
-  str1: string,
-  str2: string,
-  priorityString?: string
-) {
-  if (priorityString) {
-    if (str1 === priorityString) {
-      return -1;
-    }
-    if (str2 === priorityString) {
-      return 1;
-    }
-  }
-  return str1.localeCompare(str2);
+function insertChainConfigs(
+  networkConfigs: NetworkConfig[],
+  chainConfigs: EthereumChainConfig[]
+): NetworkConfig[] {
+  const chainConfigById = Object.fromEntries(
+    chainConfigs.map((config) => [config.id, config.value])
+  );
+  return networkConfigs.map((network) =>
+    mergeChainAndNetworkConfigs(network, chainConfigById[network.id])
+  );
 }
```

Comment:
Judging by implementation, this function does not _insert_ new values to networkConfigs, so the name is misleading
Instead it seems to update some of the `networkConfigs` with values from `chainConfigs`. Values not found in `networkConfigs` are ignored.

Reply:
Hm, why not. It is inserting chainConfigs into networkConfigs when possible.

Reply:
Ok, I renamed this :)